### PR TITLE
fix: `!server` command requirements.txt parsing

### PR DESCRIPTION
**Pull request by minisbett**
The requirements.txt has been modified recently and no longer has the simple format package==version. It also now contains lines like

`-i https://pypi.org/simple` or
`git+https://github.com/osuAkatsuki/akatsuki-pp-py@abd33445a4f48ddf69c8e7996f7e9ed7340606a0#egg=akatsuki-pp-py`

that need to be handled accordingly.
I also wanted to handle that the requirements-dev.txt is used but I don't know how to check for when this is used

EDIT: Switched to using importlib.metadata for getting the runtime dependencies

### CODE REVIEW

File: app/commands.py [hunk 1]
```diff
@@ -1256,10 +1256,22 @@ async def server(ctx: Context) -> Optional[str]:
...
     # psutil v5.8.0 | py3rijndael v0.3.3 | uvloop v0.15.2
-    reqs = (Path.cwd() / "requirements.txt").read_text().splitlines()
+    requirements = []
+
+    for line in (Path.cwd() / "requirements.txt").read_text().splitlines():
+        line = line.split(";")[0].strip()
+
+        if line.startswith("git+"):
+            split = line.split("@")[0].split("/")[-2:]
+            requirements.append(f"git:{'/'.join(split)}")
```
Comment (NiceAesth) [inline app/commands.py:1266]: Expected behaviour for git packages is to provide the `#egg={pkgname}` info

Comment (minisbett) [reply app/commands.py:1266]: what does the egg mean? I'd like to show both the github user and the repository name, only the repository name won't get you far

Comment (NiceAesth) [reply app/commands.py:1266]: `egg` is the package name of the package when installed

Comment (NiceAesth) [reply app/commands.py:1266]: see: [pip requirements format](https://pip.pypa.io/en/stable/reference/requirements-file-format/)

**Comment by NiceAesth** [comment]
https://github.com/osuAkatsuki/bancho.py/issues/335#issuecomment-1374748884 this should be of interest to you

**Comment by minisbett** [comment]
> [#335 (comment)](https://github.com/osuAkatsuki/bancho.py/issues/335#issuecomment-1374748884) this should be of interest to you

@cmyui I'm not show how you could reliably get all dependencies at runtime without checking the requirements.txt file, any ideas?

File: app/commands.py [hunk 2]
```diff
@@ -1256,10 +1256,15 @@ async def server(ctx: Context) -> Optional[str]:
     # cmyui v1.7.3 | datadog v0.40.1 | geoip2 v4.1.0
     # maniera v1.0.0 | mysql-connector-python v8.0.23 | orjson v3.5.1
     # psutil v5.8.0 | py3rijndael v0.3.3 | uvloop v0.15.2
-    reqs = (Path.cwd() / "requirements.txt").read_text().splitlines()
+    requirements = []
+
+    for dist in importlib.metadata.distributions():
+        requirements.append(f"{dist.name} v{dist.version}")
+    requirements.sort(key=lambda x: x.lower())
```
Comment (cmyui) [inline app/commands.py:1263]:
```suggestion
    requirements.sort(key=lambda x: x.casefold())
```

File: app/commands.py [hunk 3]
```diff
@@ -1256,10 +1256,15 @@ async def server(ctx: Context) -> Optional[str]:
     # cmyui v1.7.3 | datadog v0.40.1 | geoip2 v4.1.0
     # maniera v1.0.0 | mysql-connector-python v8.0.23 | orjson v3.5.1
     # psutil v5.8.0 | py3rijndael v0.3.3 | uvloop v0.15.2
-    reqs = (Path.cwd() / "requirements.txt").read_text().splitlines()
+    requirements = []
+
+    for dist in importlib.metadata.distributions():
+        requirements.append(f"{dist.name} v{dist.version}")
```
Comment (cmyui) [inline app/commands.py:1262]: i don't think this is worth fixing but the name attribute technically doesn't exist on all distribution classes
```suggestion
        requirements.append(f"{dist.name} v{dist.version}")  # type: ignore
```

**Review by cmyui — APPROVED** [review]

**Pull request merged**